modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/FilterScreen/FilterScreen.tsx

```diff
@@ -119,6 +72,87 @@ export function FilterScreen() {
 		return currentErrors;
 	};
 
+	const handleAddFilters = () => {
+		const parentWindow = Liferay.Util.getOpener();
+
+		parentWindow.Liferay.fire('openModalAddFilter', {
+			currentFilters: objectViewFilterColumns,
+			disableDateValues: true,
+			filterOperators,
+			header: Liferay.Language.get('new-filter'),
+			modalType: 'add',
+			objectFields: objectFields.filter(
+				(objectField: ObjectFieldView) => {
+					if (
+						objectField.businessType === 'Picklist' ||
+						(Liferay.FeatureFlags['LPS-152650'] &&
+							objectField.businessType === 'Relationship') ||
+						objectField.name === 'createDate' ||
+						objectField.name === 'modifiedDate' ||
+						(objectField.name === 'status' &&
+							!objectField.hasFilter)
+					) {
+						return objectField;
+					}
+				}
+			),
+			onSave: (
+				objectFieldName: string,
+				filterBy?: string,
+				fieldLabel?: LocalizedValue<string>,
+				objectFieldBusinessType?: string,
+				filterType?: string,
+				valueList?: IItem[]
+			) => {
+				dispatch({
+					payload: {
+						filterType,
+						objectFieldName,
+						valueList,
+					},
+					type: TYPES.ADD_OBJECT_VIEW_FILTER_COLUMN,
+				});
+			},
+			showModal: true,
+			validate: validateFilters,
+			workflowStatusJSONArray,
```

Comment:
Hey @rebsilva, you can create a const `filterModalConfig` here to avoid code repetition between `handleAddFilters` and `handleEditFilters`, in my tests it would look like this:

```ts
const filterModalConfig = {
    currentFilters: objectViewFilterColumns,
    disableDateValues: true,
    filterOperators,
    showModal: true,
    validate: validateFilters,
    workflowStatusJSONArray,
}
```
And the differences between the two functions from what I've seen here is the `header`, `modalType`, `objectFields` and `onSave`, so you can pass them as specific properties like:

```ts
const handleAddFilters = () => {
    const parentWindow = Liferay.Util.getOpener();
    
    parentWindow.Liferay.fire('openModalAddFilter', {
	    header: Liferay.Language.get('new-filter'),
	    modalType: 'add',
	    objectFields: objectFields.filter(
		    (objectField: ObjectFieldView) => {
			    if (
				    objectField.businessType === 'Picklist' ||
				    (Liferay.FeatureFlags['LPS-152650'] &&
					    objectField.businessType === 'Relationship') ||
				    objectField.name === 'createDate' ||
				    objectField.name === 'modifiedDate' ||
				    (objectField.name === 'status' &&
					    !objectField.hasFilter)
			    ) {
				    return objectField;
			    }
		    }
	    ),
	    onSave: (
		    objectFieldName: string,
		    filterBy?: string,
		    fieldLabel?: LocalizedValue<string>,
		    objectFieldBusinessType?: string,
		    filterType?: string,
		    valueList?: IItem[]
	    ) => {
		    dispatch({
			    payload: {
				    filterType,
				    objectFieldName,
				    valueList,
			    },
			    type: TYPES.ADD_OBJECT_VIEW_FILTER_COLUMN,
		    });
	    },
	    ...filterModalConfig
    });
};

const handleEditFilters = (objectFieldName: string) => {
    const parentWindow = Liferay.Util.getOpener();
    
    parentWindow.Liferay.fire('openModalAddFilter', {
	    editingObjectFieldName: objectFieldName,
	    modalType: 'edit',
	    objectFields,
	    onSave: (
		    objectFieldName: string,
		    filterBy?: string,
		    fieldLabel?: LocalizedValue<string>,
		    objectFieldBusinessType?: string,
		    filterType?: string,
		    valueList?: IItem[]
	    ) => {
		    dispatch({
			    payload: {
				    filterType,
				    objectFieldName,
				    valueList,
			    },
			    type: TYPES.EDIT_OBJECT_VIEW_FILTER_COLUMN,
		    });
	    },
	    ...filterModalConfig
    });
};
```